bodhi/server/consumers/masher.py

```diff
@@ -256,14 +256,16 @@ def work(self, msg):
         last_key = None
         threads = []
         for batch in batches:
-            if last_key is not None and request_order_key(batch) != last_key:
+            if ((last_key is not None and request_order_key(batch) != last_key) or
+                    (len(threads) >= int(config.get('max_concurrent_mashes', 2)))):
```

Comment:
Since we have the config validator on this setting, we don't need to convert it to an int or have a default here, so let's replace the above line with:

```
(len(threads) >= config['max_concurent_mashes'])
```